box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestDownload.java

```diff
@@ -203,60 +216,81 @@ public BoxDownload onResponse(Class clazz, BoxHttpResponse response) throws Ille
             String contentType = response.getContentType();
             long contentLength = -1;
 
-            if (response.getResponseCode() == HttpStatus.SC_OK || response.getResponseCode() == HttpStatus.SC_PARTIAL_CONTENT){
+            if (response.getResponseCode() == HttpStatus.SC_ACCEPTED) {
+                try {
+                    if (mNumRetries < DEFAULT_NUM_RETRIES) {
+                        mNumRetries++;
+                        // First try with retry-after value, then use exponential back-off
+                        String value = response.getHttpURLConnection().getHeaderField("Retry-After");
+                        if (!SdkUtils.isBlank(value)) {
+                            int retryAfterSeconds = Integer.parseInt(value);
+                            retryAfterSeconds = retryAfterSeconds > 0 ? retryAfterSeconds : 1;
+                            mRetryAfterMillis = retryAfterSeconds * 1000;
+                        }
+                    } else if (mRetryAfterMillis < DEFAULT_MAX_WAIT_MILLIS * 1000) {
+                        mRetryAfterMillis *= 2;
+                    } else {
+                        // Give up after the maximum retry time is exceeded.
+                        throw new BoxException.MaxAttemptsExceeded("Max wait time exceeded.", mNumRetries);
+                    }
+                    Thread.sleep(mRetryAfterMillis);
+                    return (BoxDownload) mRequest.send();
```

Comment:
Can you see how this works with a different executor? I'm not sure exactly what will happen here since calling mRequest.send() will create a new future task to send the data. As such I'm unsure if the original caller is getting this data in the future task they created.